diff --git a/CMakeLists.txt b/CMakeLists.txt
index 8fd185f..19a60b8 100644
--- a/CMakeLists.txt
+++ b/CMakeLists.txt
@@ -764,9 +764,14 @@ if (BUILD_TESTING)
     add_custom_target(check ${CMAKE_CTEST_COMMAND})
 endif()
 
+option(OPENSSL_BUILD_APPS "Build the OpenSSL command-line applications" ON)
+configure_file(apps/fipskey.h.cmake.in ${CMAKE_BINARY_DIR}/include/openssl/fipskey.h @ONLY)
+
 add_subdirectory(crypto)
 add_subdirectory(ssl)
-add_subdirectory(apps)
+if (OPENSSL_BUILD_APPS)
+    add_subdirectory(apps)
+endif()
 # Engines were removed in OpenSSL 4.0
 # add_subdirectory(engines)
 add_subdirectory(tools)
